From 378ca171e4d767d8ea5c666b6996a3fd25b72b58 Mon Sep 17 00:00:00 2001
From: Zheng Zucheng <zhengzucheng@huawei.com>
Date: Fri, 2 Dec 2022 09:24:15 +0800
Subject: [PATCH 0109/1398] [Huawei] fork: Fixed the extended kabi memory is
 not initialized

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 188083

-------------------------------

If the extended kabi memory is not initialized, maybe has security risks.
Therefore, the extended kabi memory is initialized to NULL in fork process
and initialized by users as required.

Fixes: dfbe33eb35fe ("[Huawei] fork: Allocate a new task_struct_resvd object for fork task")
Signed-off-by: Zheng Zucheng <zhengzucheng@huawei.com>
---
 kernel/fork.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/fork.c b/kernel/fork.c
index 4cbcaec80926..5df3cb7d01b7 100644
--- a/kernel/fork.c
+++ b/kernel/fork.c
@@ -858,7 +858,7 @@ void set_task_stack_end_magic(struct task_struct *tsk)
 static bool dup_resvd_task_struct(struct task_struct *dst,
 				  struct task_struct *orig, int node)
 {
-	dst->_resvd = kmalloc_node(sizeof(struct task_struct_resvd),
+	dst->_resvd = kzalloc_node(sizeof(struct task_struct_resvd),
 					  GFP_KERNEL, node);
 	if (!dst->_resvd)
 		return false;
-- 
2.17.1

